ui/components-react/src/test/propertygrid/component/internal/flat-items/CategorizedArrayProperty.test.ts

```diff
@@ -176,6 +176,81 @@ describe("CategorizedArrayProperty", () => {
         expect(overrideDisplayLabel).toEqual(expectedOverrideDisplayLabel);
       });
     });
+
+    it("Should append property description to non-primitive items that have more that 1 items", () => {
+      const propertyRecord = TestUtils.createArrayProperty(
+        "CADID1",
+        [
+          TestUtils.createPrimitiveStringProperty("CADID1_1", "V1"),
+          TestUtils.createStructProperty("CADID1_6", {
+            prop1: TestUtils.createPrimitiveStringProperty(
+              "CADID1_6_1",
+              "test"
+            ),
+            prop2: TestUtils.createPrimitiveStringProperty(
+              "CADID1_6_2",
+              "test"
+            ),
+          }),
+          TestUtils.createArrayProperty("CADID1_7", [
+            TestUtils.createPrimitiveStringProperty("CADID1_7_1", "test"),
+            TestUtils.createPrimitiveStringProperty("CADID1_7_2", "test"),
+          ]),
+        ].map((record, idx) => {
+          record.description = `${idx}`;
```

Comment:
Maybe make this `Description_${idx}`? While here it's pretty clear what you're doing, down below, where you do the checks, the assertion is quite confusing: `[${index + 1}] ${index}`.

Reply:
Done